samcli/lib/runtimes/base.py

```diff
@@ -0,0 +1,342 @@
+        Family.JAVA,
+        "amazon/java21-base",
+        [Architecture.X86_64, Architecture.ARM64],
+    )
+    java17 = (
+        "java17",
+        Family.JAVA,
+        "amazon/java17-base",
+        [Architecture.X86_64, Architecture.ARM64],
+    )
+    java11 = (
+        "java11",
+        Family.JAVA,
+        "amazon/java11-base",
+        [Architecture.X86_64, Architecture.ARM64],
+    )
+    java8al2 = (
+        "java8.al2",
+        Family.JAVA,
+        "amazon/java8.al2-base",
+        [Architecture.X86_64, Architecture.ARM64],
+    )
+    java8 = (
+        "java8",
+        Family.JAVA,
+        "amazon/java8-base",
+        [Architecture.X86_64],
+    )
+
+    nodejs20x = (
+        "nodejs20.x",
+        Family.NODEJS,
+        "amazon/nodejs20.x-base",
+        [Architecture.X86_64, Architecture.ARM64],
+    )
+    nodejs18x = (
+        "nodejs18.x",
+        Family.NODEJS,
+        "amazon/nodejs18.x-base",
+        [Architecture.X86_64, Architecture.ARM64],
+    )
+    nodejs16x = (
+        "nodejs16.x",
+        Family.NODEJS,
+        "amazon/nodejs16.x-base",
+        [Architecture.X86_64, Architecture.ARM64],
+    )
+
+    provided_al2023 = (
+        "provided.al2023",
+        Family.PROVIDED,
+        None,
+        [Architecture.X86_64, Architecture.ARM64],
+    )
+    provided_al2 = (
+        "provided.al2",
+        Family.PROVIDED,
+        None,
+        [Architecture.X86_64, Architecture.ARM64],
+    )
+    provided = (
+        "provided",
+        Family.PROVIDED,
+        None,
+        [Architecture.X86_64],
+    )
+    go_provided_al2023 = (
+        "go (provided.al2023)",
+        Family.PROVIDED,
+        "amazon/go-provided.al2023-base",
+        [Architecture.X86_64, Architecture.ARM64],
+        False,
+    )
+    go_provided_al2 = (
+        "go (provided.al2)",
+        Family.PROVIDED,
+        "amazon/go-provided.al2-base",
+        [Architecture.X86_64, Architecture.ARM64],
+        False,
+    )
+
+    python312 = (
+        "python3.12",
+        Family.PYTHON,
+        "amazon/python3.12-base",
+        [Architecture.X86_64, Architecture.ARM64],
+    )
+    python311 = (
+        "python3.11",
+        Family.PYTHON,
+        "amazon/python3.11-base",
+        [Architecture.X86_64, Architecture.ARM64],
+    )
+    python310 = (
+        "python3.10",
+        Family.PYTHON,
+        "amazon/python3.10-base",
+        [Architecture.X86_64, Architecture.ARM64],
+    )
+    python39 = (
+        "python3.9",
+        Family.PYTHON,
+        "amazon/python3.9-base",
+        [Architecture.X86_64, Architecture.ARM64],
+    )
+    python38 = (
+        "python3.8",
+        Family.PYTHON,
+        "amazon/python3.8-base",
+        [Architecture.X86_64, Architecture.ARM64],
+    )
+
+    ruby32 = (
+        "ruby3.2",
+        Family.RUBY,
+        "amazon/ruby3.2-base",
+        [Architecture.X86_64, Architecture.ARM64],
+    )
+
+
+@unique
+class DeprecatedRuntime(RuntimeEnumBase):
```

Comment:
When a runtime deprecates, we move it into this list by just preserving the first two fields?

Reply:
yep, that's right